src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -639,6 +612,184 @@ save_availableCaptureId > (testExpressionCaptureId = ((IFlowCaptureReferenceOper
+                               // PROTOTYPE(dataflow): The scenario with missing GetValueOrDefault is not covered by unit-tests.
+                               MakeInvalidOperation(((INamedTypeSymbol)testExpressionType).TypeArguments[0], receiver);
+                }
+
+                Debug.Assert(_currentConditionalAccessInstance == null);
+                _currentConditionalAccessInstance = receiver;
+
+                if (currentConditionalAccess.WhenNotNull.Kind != OperationKind.ConditionalAccess)
+                {
+                    break;
+                }
+
+                currentConditionalAccess = (IConditionalAccessOperation)currentConditionalAccess.WhenNotNull;
+            }
+
+            int resultCaptureId = captureIdForResult ?? _availableCaptureId++;
+
+            if (IsNullableType(operation.Type) && !IsNullableType(currentConditionalAccess.WhenNotNull.Type))
+            {
+                IOperation access = Visit(currentConditionalAccess.WhenNotNull);
+                AddStatement(new FlowCapture(resultCaptureId, currentConditionalAccess.WhenNotNull.Syntax,
+                                             TryMakeNullableValue(operation.Type, access) ??
+                                             // PROTOTYPE(dataflow): The scenario with missing constructor is not covered by unit-tests.
+                                             MakeInvalidOperation(operation.Type, access)));
+            }
+            else
+            {
+                VisitAndCapture(currentConditionalAccess.WhenNotNull, resultCaptureId);
+            }
+
+            Debug.Assert(_currentConditionalAccessInstance == null);
+
+            var afterAccess = new BasicBlock(BasicBlockKind.Block);
+            LinkBlocks(CurrentBasicBlock, afterAccess);
+            _currentBasicBlock = null;
+
+            AppendNewBlock(whenNull);
+
+            SyntaxNode defaultValueSyntax = (whenNull.Predecessors.Count > 1 ? operation : currentConditionalAccess.Operation).Syntax;
+
+            AddStatement(new FlowCapture(resultCaptureId,
+                                         defaultValueSyntax,
+                                         new DefaultValueExpression(semanticModel: null, defaultValueSyntax, operation.Type,
+                                                                    (operation.Type.IsReferenceType && !IsNullableType(operation.Type)) ?
+                                                                        new Optional<object>(null) : default,
+                                                                    isImplicit: true)));
+
+            AppendNewBlock(afterAccess);
+
+            return new FlowCaptureReference(resultCaptureId, operation.Syntax, operation.Type, operation.ConstantValue);
+        }
+
+        private IOperation TryMakeNullableValue(ITypeSymbol type, IOperation underlyingValue)
+        {
+            Debug.Assert(IsNullableType(type));
+
+            foreach (ISymbol candidate in type.GetMembers(".ctor"))
```

Comment:
Consider changing `type` to `INamedTypeSymbol` and loop over `type.Constructors`. Then some of the method candidate checks below can be removed.